modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/near/GridNearTxLocal.java

```diff
@@ -4316,7 +4317,12 @@ public IgniteInternalFuture<GridNearTxPrepareResponse> prepareAsyncLocal(GridNea
             userPrepare((serializable() && optimistic()) ? F.concat(false, req.writes(), req.reads()) : req.writes());
 
             // Make sure to add future before calling prepare on it.
-            cctx.mvcc().addFuture(fut);
+            GridCacheMvccManager mvcc = cctx.mvcc();
+
+            if (mvcc == null)
+                throw stoppingException();
+            else
+                cctx.mvcc().addFuture(fut);
```

Comment:
It should be `mvvc.addFuture(fut)` I suppose.

Reply:
sure, thanks!